### PR TITLE
Added synthetic traffic filtering to Microsoft.AI.Web

### PR DESCRIPTION
**Pull request by madeleinefletcher**

### CODE REVIEW

**Comment by madeleinefletcher** [comment]
@abaranch @SergeyKanzhelev @AlexBulankou

File: Src/Web/Web.Nuget/Resources/ApplicationInsights.config.install.xdt [hunk 1]
```diff
@@ -25,14 +36,10 @@
          xdt:Transform="InsertIfMissing"
          xdt:Locator="Match(Type)">
       <UserAgents>
```
Comment (abaranch) [inline Src/Web/Web.Nuget/Resources/ApplicationInsights.config.install.xdt]: Let's completely remove TP

File: Src/Web/Web.Shared.Net/Implementation/WebEventSource.cs [hunk 2]
```diff
@@ -270,7 +270,19 @@ public void WebRequestFilteredOutByRequestHandler(string appDomainName = "Incorr
                 30,
                 this.ApplicationName);
         }
-      
+
+        [Event(
+            31,
+            Message = "[msg=SyntheticUserAgentTelemetryInitializerRegularExpressionParsingException];[exception={0}];",
```
Comment (abaranch) [inline Src/Web/Web.Shared.Net/Implementation/WebEventSource.cs]: Looks like a user actionable error. Let's add more user friendly message and add UserActionable keyword

File: Src/Web/Web.Shared.Net/SyntheticUserAgentFilter.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+{
+    using System;
+    using System.Text.RegularExpressions;
+    using Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+    /// <summary>
+    /// Allows configuration of patterns for synthetic traffic filters.
+    /// </summary>
+    public class SyntheticUserAgentFilter
```
Comment (abaranch) [inline Src/Web/Web.Shared.Net/SyntheticUserAgentFilter.cs]: Do we need it to be public?

Comment (madeleinefletcher) [inline Src/Web/Web.Shared.Net/SyntheticUserAgentFilter.cs]: We do if we want to support configuration through code...

File: Src/Web/Web.Shared.Net/Implementation/WebEventSource.cs (hunk 2 above)
Comment (madeleinefletcher) [inline Src/Web/Web.Shared.Net/Implementation/WebEventSource.cs]: Fixed.

File: Src/Web/Web.Nuget/Resources/ApplicationInsights.config.install.xdt (hunk 1 above)
Comment (madeleinefletcher) [inline Src/Web/Web.Nuget/Resources/ApplicationInsights.config.install.xdt]: Fixed.

**Comment by abaranch** [comment]
:shipit:

**Pull request merged**